java/java.hints/src/org/netbeans/modules/java/hints/suggestions/Tiny.java

```diff
@@ -400,4 +402,83 @@ protected void performRewrite(final TransformationContext ctx) {
 
     }
 
+    @Hint(displayName = "#DN_org.netbeans.modules.java.hints.suggestions.Tiny.inlineRedundantVar", description = "#DESC_org.netbeans.modules.java.hints.suggestions.Tiny.inlineRedundantVar", category="suggestions", hintKind=Kind.ACTION, severity=Severity.HINT)
+    @TriggerPattern(value="$type $var = $init; $statements$ return $var;")
```

Comment:
the trigger seems to be a bit greedy. The hint is active even before the variable is declared.
![inline](https://user-images.githubusercontent.com/114367/175393374-c700689b-fc28-4914-83bd-a45fe1a560fb.png)